front/src/applications/editor/tools/translationTools.ts

```diff
@@ -0,0 +1,79 @@
+import i18n from 'i18n';
+import { TFunction } from 'i18next';
+import { JSONSchema7, JSONSchema7Definition } from 'json-schema';
+
+export function translateProperties(
+  propertiesList: {
+    [key: string]: JSONSchema7Definition;
+  },
+  defName: string,
+  t: TFunction
+) {
+  return Object.keys(propertiesList).reduce(
+    (propertiesAcc, property) => {
+      const currentProperty = propertiesList[property] as JSONSchema7;
+      const currentDescriptionKey = `${defName}.properties.${property}.description`;
+      const currentTitleKey = `${defName}.properties.${property}.title`;
+
+      return {
+        ...propertiesAcc,
+        [property]: {
+          ...currentProperty,
+          ...(currentProperty.description && {
+            description: i18n.exists(`infraEditor:${currentDescriptionKey}`)
+              ? t(`infraEditor:${currentDescriptionKey}`)
+              : currentProperty.description,
```

Comment:
We can extract the logic of use default translation if the key doesn't exist in translation dictionary and move it to a function like:
```
function translate(key:string, defaultTranslation){
  return i18n.exists(key) ? t(key) : defaultTranslation
}
```
And then you can use it like :
`{ description : translate('infraEditor:${currentDescriptionKey}', currentProperty.description) }`

Reply:
Good idea, Thanks !